test/agent-policy-tests/test-script-utils.sh

```diff
@@ -0,0 +1,395 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+####################################################################
+## This script contains unit tests for
+## modules/agent-policy/scripts/script-utils.sh
+## This script is ran via test/agent-policy-tests/test-runner.sh
+####################################################################
+
+PROJECT_ID="test-project-id"
+POLICY_ID="ops-agents-test-policy"
+DESCRIPTION="an example test policy"
+AGENT_RULES_JSON_BASIC="[{\"type\":\"metrics\"}]"
+OS_TYPES_JSON_BASIC="[{\"short_name\":\"centos\",\"version\":\"8\"}]"
+EMPTY_LIST_JSON="[]"
+ETAG="db5c5a61-6a05-4f67-8f84-c89a654eb576"
+
+
+# include functions to build gcloud command
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+UTILS_ABS_PATH="${SCRIPT_DIR}/../../modules/agent-policy/scripts/script-utils.sh"
+# shellcheck disable=SC1090
+source "$UTILS_ABS_PATH"
+
+
+##############################################################
+##  util functions                                          ##
+##############################################################
+
+# Params:
+#   $1 = test name
+#   $2 = expected (string)
+#   $3 = actual (string)
+# Compares the expected and actual values. Prints the result
+# of the comparison, and prints a diff if the expected and actual
+# values are differet.
+function assert_equals() {
+    local test_name="$1"
+    local expected="$2"
+    local actual="$3"
+    # assert that the expected and actual are equal
+    if [ "$expected" = "$actual" ]; then
+        echo "===== $test_name: SUCCESS"
+    else
+        echo "===== $test_name: FAILURE"
+        echo "=====   expected:"
+        echo "=====      \"$expected\""
+        echo "=====   actual:"
+        echo "=====      \"$actual\""
+    fi
+}
+
+
+
+##############################################################
+##  get_create_command tests                                ##
+##############################################################
+
+function test_get_create_command_defaults() {
+    local description=""
+    local agent_rules_json="$AGENT_RULES_JSON_BASIC"
+    local group_labels_json="$EMPTY_LIST_JSON"
+    local os_types_json="$OS_TYPES_JSON_BASIC"
+    local zones_json="$EMPTY_LIST_JSON"
+    local instances_json="$EMPTY_LIST_JSON"
+    local expected_command
+    local actual_command
+
+    expected_command="gcloud alpha compute instances ops-agents"
+    expected_command="$expected_command policies create ops-agents-test-policy"
+    expected_command="$expected_command --agent-rules='type=metrics'"
+    expected_command="$expected_command --os-types='version=8,short-name=centos'"
+    expected_command="$expected_command --project='test-project-id' --quiet"
```

Comment:
Why are you building up the expected command over multiple lines (same question throughout this script)?

Reply:
If I remember correctly, I was running into issues with the linter + getting the spacing exactly right, so building up the expected command like this was the way I was able to preserve the right spacing without linter errors.